File: problems/665/solution.cpp
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int idx = -1;
        
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] > nums[i + 1]) {
                if (idx == -1) {
                    idx = i;
                } else {
                    return false;
                }
            }
        }
        
        if (idx <= 0 or idx >= nums.size() - 2) {
            return true;
        }
        
        return nums[idx - 1] <= nums[idx + 1] or nums[idx] <= nums[idx + 2];
    }
};

File: problems/665/solution.go
package main
func checkPossibility(nums []int) bool {
    p := -1
    
    for i := 0; i < len(nums) - 1; i++ {
        if nums[i] > nums[i + 1] {
            if p != -1 {
                return false
            }
            
            p = i
        }
    }
    
    return p == -1 || p == 0 || p == len(nums) - 2 || nums[p - 1] <= nums[p + 1] || nums[p] <= nums[p + 2]
}